src/common/components/elements/NFTPreviewGrid.tsx

```diff
@@ -76,49 +74,162 @@ const getOverlayStatus = (index: number, nftValues?: NFTValue[]) => {
 
 const StyledAntDImage = styled(AntImage)`
   object-fit: cover;
+  width: 120px;
+  height: 120px;
 `;
 
+const StyledPreviewOverlay = styled.div`
+  position: relative;
+  max-height: 120px; // Margin collapse really being annoying
+`;
+
+const PrevSvgWrapper = styled.div`
+  position: absolute;
+  background: black;
+  border-radius: 15px;
+  bottom: 13px;
+  left: 6px;
+  z-index: 10;
+  height: 25%;
+  width: 25%;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const PreviewOverlay = ({ children, icon }: { children: React.ReactElement; icon: string }) => {
+  return (
+    <StyledPreviewOverlay>
+      <PrevSvgWrapper>
+        <FeatherIcon icon={icon} />
+      </PrevSvgWrapper>
+
+      {children}
+    </StyledPreviewOverlay>
+  );
+};
+
+interface Props {
+  filePreviews: Array<FilePreview>;
+  index?: number;
+  width?: number;
+  children?: any;
+  nftValues?: NFTValue[];
+}
+
+const Preview = ({ onClick, fp, icon }: { onClick: () => void; fp: FilePreview; icon: string }) => {
+  return (
+    <StyledPrevWrapper onClick={onClick}>
+      {fp.coverImage ? (
+        <PreviewOverlay icon={icon}>
+          <StyledAntDImage
+            src={URL.createObjectURL(fp.coverImage)}
+            alt={fp.file.name}
+            preview={false}
+          />
+        </PreviewOverlay>
+      ) : (
+        <FeatherIcon icon={icon} />
+      )}
+    </StyledPrevWrapper>
+  );
+};
+
 export const NFTPreviewGrid = ({
-  images,
+  filePreviews,
   index = -1,
   width = 2,
-  removeImage,
   children,
-  isMintStep,
   nftValues,
 }: Props) => {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [currentFile, setCurrentFile] = useState<File>();
+  const showVidAudPreview = currentFile && (isVideo(currentFile) || isAudio(currentFile));
+  const show3DFilePreview = currentFile && is3DFile(currentFile);
+
+  const vidRef = useRef<HTMLVideoElement>(null);
+
+  const showModal = () => {
+    setIsModalVisible(true);
+  };
+
+  const handleOk = () => {
+    setIsModalVisible(false);
+
+    if (vidRef.current) {
+      vidRef.current.pause();
+    }
+  };
+
+  const handlePrevClick = (fp: FilePreview) => {
+    setCurrentFile(fp.file);
+    showModal();
+  };
+
+  const getFilePreview = (fp: FilePreview) => {
+    if (isAudio(fp)) {
+      return <Preview onClick={() => handlePrevClick(fp)} fp={fp} icon="volume-2" />;
+    }
+
+    if (isVideo(fp)) {
+      return <Preview onClick={() => handlePrevClick(fp)} fp={fp} icon="youtube" />;
```

Comment:
There isn't a general video play icon? Don't want to use youtube as our play button.

Reply:
@damianr Requested `youtube over `video`.  What's the concern?